jest-shared-setup.ts

```diff
@@ -2,7 +2,15 @@ import failOnConsole from "jest-fail-on-console"
 import '@testing-library/jest-dom'
 import { configure } from "@testing-library/dom"
 
-failOnConsole()
+failOnConsole({
+  /**
+   * This warning is removed from newer versions of React because it is
+   * misleading (not usually a memory leak).
+   *
+   * See https://github.com/reactwg/react-18/discussions/82
+   */
+  silenceMessage: msg => /Can't perform a React state update on an unmounted component/.test(msg),
```

Comment:
This warning has been removed in React 18 because it is misunderstood and misleading (generally not a memory leak).

React Query [recommends](https://github.com/TanStack/query/issues/3804#issuecomment-1184212985) ignoring these warnings since they are removed from react anyway (in newer versions, as mentioned).